### PR TITLE
Improve diff output and unify lint/fix code paths

### PR DESCRIPTION
**Pull request by snarlysodboxer**
## Summary

- Replace the raw text diff with a structural YAML-like summary that shows what moved and what was added, making it much easier to understand what changed at a glance
- Unify lint and fix to use the same code path (`WalkAndSort` + `ComputeDescriptions`), removing `WalkAndCompare` and `checkPairs` (~700 lines)
- Fix a bug where comments above `---` document markers could get duplicated
- Replace godebug/diff with gotextdiff for colored unified diff output (available on demand)
- Add external diff tool support with auto-detection of common tools

## Structural output summary

Both lint and fix now show a YAML-like tree of changes instead of flat error strings or a raw diff:

File: deployment.yaml

```yaml
  Changes:
    metadata:
      name: cool-app  # move to top
      namespace: default  # move up
    spec:
      template:
        spec:
          initContainers[0]:
            name: wait-for-something  # move to top
          containers[0]:
            env: [...]  # move up
        metadata:
          labels:
            app.kubernetes.io/name: TODO  # add
```

Scalar values are shown inline, maps as {...}, sequences as [...]. Moved keys are annotated in green, added keys in bold yellow (when output is a terminal).

Interactive prompt: The `fix` command now prompts with options to view a full diff on demand.

External diff tool: The `e` option opens the before/after in an external diff tool.

## Code path unification

Lint previously used `WalkAndCompare` (read-only validation) while fix used WalkAndSort (reordering). These were structurally identical except at the mapping node level, creating a
maintenance burden and divergence risk. Now both commands use `WalkAndSort`.

## Bug fix

Fixed comment duplication when `fixHeadComment` and `fixFootComment` couldn't find where a comment went in the new tree (e.g., comments above `---`).

## Added comment preservation test suite:

Added pkg/whitespace/comments_edge_cases_test.go with:
- 17 edge case tests for PreserveComments
- 5 full-pipeline tests (PreserveComments + PreserveEmptyLines together)
- 3 yaml.v3 round-trip behavior tests
- Bug fix regression tests
- Known limitation documentation tests

### CODE REVIEW

File: cmd/fix.go [hunk 1]
```diff
@@ -127,21 +132,25 @@ var fixCmd = &cobra.Command{
...
-				errs := compare.WalkAndCompare(configNode, fileNode, sortConfigs, compare.ValidationErrors{})
-				if len(errs) == 0 {
-					continue
-				}
+			// parse a separate copy for move summary comparison
+			oldFileNode, err := parseNodeFromBytes(existingFileContents)
+			if err != nil {
+				log.Fatalf("error parsing yaml for target file: %s: %v", filePath, err)
 			}
+			commentCount := moves.CountComments(oldFileNode)
```
[the remark this answers is not in the document]
Comment (snarlysodboxer) [reply cmd/fix.go:144]: fixed

File: cmd/fix.go [hunk 2]
```diff
@@ -253,6 +264,142 @@ func init() {
...
+	newFile.Close()
+	if err := os.Chmod(newFile.Name(), 0444); err != nil {
+		return fmt.Errorf("setting temp file read-only: %w", err)
+	}
+
+	// Split command, respecting that the tool may have arguments
+	parts := strings.Fields(tool)
+	args := append(parts[1:], oldFile.Name(), newFile.Name())
+
+	cmd := exec.Command(parts[0], args...)
```
[the remark this answers is not in the document]
Comment (snarlysodboxer) [reply cmd/fix.go:392]: fixed

File: cmd/fix.go [hunk 3]
```diff
@@ -264,23 +411,42 @@ func countLines(str string, separator rune) int {
...
+		case "d":
+			fmt.Printf("\n%s\n", generateDiff(filePath, oldContent, newContent))
+		case "e":
+			tool := getExternalDiffTool()
+			if tool == "" {
+				fmt.Println("No external diff tool configured. Set one of these environment variables:")
+				fmt.Println("  PREDICTABLE_YAML_DIFF=\"vimdiff\"")
+				fmt.Println("  KUBECTL_EXTERNAL_DIFF=\"code --diff --wait\"")
+				fmt.Println("  DIFFTOOL=\"difftastic\"")
+			} else {
```
[the remark this answers is not in the document]
Comment (snarlysodboxer) [reply cmd/fix.go:451]: fixed

File: cmd/lint.go [hunk 4]
```diff
@@ -89,22 +90,42 @@ var lintCmd = &cobra.Command{
...
+				if err != nil {
+					log.Fatalf("error parsing yaml for target file: %s: %v", filePath, err)
+				}
+
+				descriptions := moves.ComputeDescriptions(oldFileNode, fileNode)
+				summary := moves.FormatSummary(filePath, descriptions, addedFields, 0)
+				if summary != "" {
+					fmt.Print("\n" + summary + "\n")
+				}
 			}
```
[the remark this answers is not in the document]
Comment (snarlysodboxer) [reply cmd/lint.go:130]: fixed
